source/game_sa/ConversationForPed.cpp

```diff
@@ -7,13 +7,100 @@ void CConversationForPed::InjectHooks() {
     RH_ScopedCategory("Conversations");
 
     RH_ScopedInstall(Clear, 0x43A770);
-    RH_ScopedInstall(Update, 0x43C190, {.reversed = false});
+    RH_ScopedInstall(Update, 0x43C190);
     RH_ScopedInstall(IsPlayerInPositionForConversation, 0x43AC40);
 }
 
 // 0x43C190
 void CConversationForPed::Update() {
+    if (!m_Ped || !m_bEnabled || !IsPlayerInPositionForConversation(false)) {
+        m_Status = 0;
+        return;
+    }
+
+    const auto pad = CPad::GetPad();
+    if (CTimer::GetTimeInMS() > m_LastTimeWeWereCloseEnough + 1000 || m_Status == 1 &&
+        (CTimer::GetTimeInMS() > m_LastChange + 4000 || pad->ConversationYesJustDown() || pad->ConversationNoJustDown())) {
```

Comment:
```suggestion
    if (   CTimer::GetTimeInMS() > m_LastTimeWeWereCloseEnough + 1000
        || m_Status == 1 && (CTimer::GetTimeInMS() > m_LastChange + 4000 || pad->ConversationYesJustDown() || pad->ConversationNoJustDown())
    ) {
```
Perhaps add `CPad::ConversationYesOrNoJustDown`